### PR TITLE
Added docs on auth setup

### PR DESCRIPTION
**Pull request by JoeStead**
I've added these in the hope that we can get a better grasp of what is required to be configured in Ironclad for authentication to behave as we want it to. It should also help come up with more fitting names for the different moving parts.

The way the `UserClaims` are configured in API scopes still confuses me a little, but I understand why API Resource Scopes exist and how they differ from clients at least (I think)

### CODE REVIEW

**Comment by JoeStead** [comment]
I've also purposefully not named the `ConfiguredApi` section and just linked to the document. Currently it's called `api` that won't scale for the other services, and calling it `apollo_api` is just confusing because we have a client named that

File: docs/Authentication/Clients.md [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+## Apollo API Client (`apollo_api`)
+`apollo_api` requires the following settings to be configured:
+
+### Allowed Grant Types
+The type of authentication for the API to talk to the Ironclad API is:
+
+* ClientCredentials
+
+**N.B.** Using `ClientCredentials` requires a `ClientId` and `Secret` to be set fo rthe client
```
Comment (jchannon) [inline docs/Authentication/Clients.md]: typo

Comment (JoeStead) [reply docs/Authentication/Clients.md]:
```suggestion
**N.B.** Using `ClientCredentials` requires a `ClientId` and `Secret` to be set for the client
```

Comment (JoeStead) [reply docs/Authentication/Clients.md]: Could've at least suggested the fix :)

**Comment by JoeStead** [comment]
That's based on `AllowedGrantTypes` right?

**Comment by yreynhout-lykke** [comment]
You need to specify it up front when you define them using the ironclad CLI. See the docker-composed PR run-local.sh script.

**Comment by JoeStead** [comment]
I think a couple of things will require an `OIDC -> Ironclad` translation? I need to familiarise my self with it some more

**Pull request merged**